examples/aligned_accessor/aligned_accessor.cpp

```diff
@@ -0,0 +1,714 @@
+#  define _MDSPAN_ASSUME_ALIGNED( ELEMENT_TYPE, POINTER, BYTE_ALIGNMENT ) reinterpret_cast< ELEMENT_TYPE* >(__builtin_assume_aligned( POINTER, BYTE_ALIGNMENT ))
+  constexpr char assume_aligned_method[] = "__builtin_assume_aligned";
+#else
+#  define _MDSPAN_ASSUME_ALIGNED( ELEMENT_TYPE, POINTER, BYTE_ALIGNMENT ) POINTER
+  constexpr char assume_aligned_method[] = "(none)";
+#endif
+
+// Some compilers other than Clang or GCC like to define __clang__ or __GNUC__.
+// Thus, we order the tests from most to least specific.
+#if defined(__ICL)
+#  define _MDSPAN_ALIGN_VALUE_ATTRIBUTE( BYTE_ALIGNMENT ) __declspec(align_value( BYTE_ALIGNMENT ));
+  constexpr char align_attribute_method[] = "__declspec(align_value(BYTE_ALIGNMENT))";
+#elif defined(__ICC)
+#  define _MDSPAN_ALIGN_VALUE_ATTRIBUTE( BYTE_ALIGNMENT ) __attribute__((align_value( BYTE_ALIGNMENT )));
+  constexpr char align_attribute_method[] = "__attribute__((align_value(BYTE_ALIGNMENT)))";
+#elif defined(__clang__)
+#  define _MDSPAN_ALIGN_VALUE_ATTRIBUTE( BYTE_ALIGNMENT ) __attribute__((align_value( BYTE_ALIGNMENT )));
+  constexpr char align_attribute_method[] = "__attribute__((align_value(BYTE_ALIGNMENT)))";
+#else
+#  define _MDSPAN_ALIGN_VALUE_ATTRIBUTE( BYTE_ALIGNMENT )
+  constexpr char align_attribute_method[] = "(none)";
+#endif
+
+template<class ElementType>
+constexpr bool
+valid_byte_alignment(const std::size_t byte_alignment,
+		     const std::size_t size_of_element)
+{
+  return byte_alignment >= alignof(ElementType); // && is a power of two
+}
+
+template<class ElementType, std::size_t byte_alignment>
+struct aligned_pointer {
+  // x86-64 ICC 2021.5.0 emits warning #3186 ("expected typedef declaration") here.
+  // No other compiler (including Clang, which has a similar type attribute) has this issue.
+  using type = ElementType* _MDSPAN_ALIGN_VALUE_ATTRIBUTE( byte_alignment );
+};
+
+template<class ElementType, std::size_t byte_alignment>
+using aligned_pointer_t = typename aligned_pointer<ElementType, byte_alignment>::type;
+
+template<class ElementType, std::size_t byte_alignment>
+aligned_pointer_t<ElementType, byte_alignment>
+bless(ElementType* ptr, std::integral_constant<std::size_t, byte_alignment> /* ba */ )
+{
+  return _MDSPAN_ASSUME_ALIGNED( ElementType, ptr, byte_alignment );
+}
+
+template<class ElementType, std::size_t byte_alignment>
+struct aligned_accessor {
+  using offset_policy = stdex::default_accessor<ElementType>;
+  using element_type = ElementType;
+  using reference = ElementType&;
+  using data_handle_type = aligned_pointer_t<ElementType, byte_alignment>;
+
+  constexpr aligned_accessor() noexcept = default;
+
+  MDSPAN_TEMPLATE_REQUIRES(
+    class OtherElementType,
+    std::size_t other_byte_alignment,
+    /* requires */ (std::is_convertible<OtherElementType(*)[], element_type(*)[]>::value && other_byte_alignment == byte_alignment)
+    )
+  constexpr aligned_accessor(aligned_accessor<OtherElementType, other_byte_alignment>) noexcept {}
```

Comment:
Why a requires clause rather than a static_assert?

Reply:
@dalg24 The `element_type` part of this imitates `default_accessor`.  It felt weird to treat byte alignment differently than `element_type`.  I won't insist, though ; - ) .

The byte alignment part could, in theory, permit conversions to lesser alignment.

Reply:
@dalg24 Please let me know if you still want this changed; thanks!